src/mailroom.py

```diff
@@ -0,0 +1,339 @@
+            CURRENT_STATE = STATE_DICT['state_4']
+        except KeyError:
+            STATE_DICT['state_2']['user_response'] = a
+            print('{} is not a current donor.'.format(a))
+            print('Create a new donor {}'.format(a), end='')
+            STATE_DICT['state_6']['new_donor_name'] = a
+            CURRENT_STATE = STATE_DICT['state_6']
+            return
+
+
+def state_3_valid_responses(a):
+    global CURRENT_STATE
+    if a == '1':
+        CURRENT_STATE = STATE_DICT['state_1']
+        os.system('clear')
+    elif a == '0':
+        os.system('clear')
+        print('Good By')
+        sys.exit()
+    else:
+        print('That was not a valid input')
+        return False
+
+
+def state_4_valid_responses(a):
+    ''' need to test for properly formated $ amout iiiiii.ii '''
+    global CURRENT_STATE
+    global CURRENT_DONOR
+    if a == '0':
+        CURRENT_STATE = STATE_DICT['state_2']
+        os.system('clear')
+        return
+    try:
+        amount = float(a)
+        print('Is ${:,.2f} correct?'.format(amount), end='')
+        STATE_DICT['state_4']['donation_amount'] = amount
+        CURRENT_STATE = STATE_DICT['state_7']
+        return amount
+    except ValueError:
+        print('{} is not a valid amount'.format(a))
+        return False
+
+
+def state_5_valid_responses(a):
+    global CURRENT_STATE
+    if a == 'y':
+        print_letter()
+        CURRENT_STATE = STATE_DICT['state_2']
+    elif a == 'n':
+        os.system('clear')
+        CURRENT_STATE = STATE_DICT['state_2']
+    else:
+        print('That was not a valid input')
+        return False
+    return
+
+
+def state_6_valid_responses(a):
+    global CURRENT_STATE
+    if a == 'y':
+        create_donor()
+        # update_donor()
+        CURRENT_STATE = STATE_DICT['state_4']
+    elif a == 'n':
+        os.system('clear')
+        CURRENT_STATE = STATE_DICT['state_2']
+    else:
+        print('That was not a valid input')
+        CURRENT_STATE = STATE_DICT['state_6']
+        return False
+    return
+
+
+def state_7_valid_responses(a):
+    global CURRENT_STATE
+    if a == 'y':
+        update_donor(STATE_DICT['state_4']['donation_amount'])
+        user_report()
+        CURRENT_STATE = STATE_DICT['state_5']
+    elif a == 'n':
+        os.system('clear')
+        CURRENT_STATE = STATE_DICT['state_4']
+    elif a == '0':
+        os.system('clear')
+        CURRENT_STATE = STATE_DICT['state_2']
+    else:
+        print('That was not a valid input')
+        CURRENT_STATE = STATE_DICT['state_7']
+        return False
+    return
+
+
+def create_report():
+    global DONORS
+    o = OrderedDict(sorted(DONORS.items(),
+                    key=lambda t: t[1]['total_donation']))
+    os.system('clear')
+    print('{:<20} {:<15}  {:<7} {:<15}  {:<15}\n'
+          .format('Name', 'Total', 'Number',
+                  'Average', 'Last'))
+    for donor in o.values():
+        print('{:<20} ${:<15,.2f} {:<7} ${:<15,.2f} ${:<15,.2f}'
```

Comment:
Remember that you can provide named placeholders, which work especially nicely when you have data in dictionaries like you do.  Consider this change to your format string:

```python
msg = '{name:<20} ${total_donation:<15,.2f} {num_donations:<7} ${avg_donation:<15,.2f} ${last_donation:<15,.2f}'
```
Given the data present in each "donor" dict, you could then modify your code like so:

```python
for donor in o.values():
    print(msg.format(**donor))
```

This is quite a bit easier to read.